src/Eloquent/Model.php

```diff
@@ -319,6 +319,34 @@ protected function fromDecimal($value, $decimals)
         return new Decimal128($this->asDecimal($value, $decimals));
     }
 
+    /**
+     * Convert MongoDB objects to their string representations.
+     *
+     * This method converts MongoDB related objects (ObjectID, Binary, UTCDateTime)
+     * to their serialized representations, ensuring the values can be correctly
+     * serialized when the model is converted to JSON.
+     *
+     * @param mixed $value The value to convert
+     *
+     * @return mixed
+     */
+    protected function convertMongoObjects(mixed $value): mixed
+    {
+        if ($value instanceof ObjectID) {
+            $value = (string) $value;
+        } elseif ($value instanceof Binary) {
+            $value = (string) $value->getData();
+        } elseif ($value instanceof UTCDateTime) {
```

Comment:
I don't think dates should be handled here. They were not in the current code.

Reply:
I understand, indeed if relationships are called, casts are automatic. Sorry for the misunderstanding.
You can reject the fix proposal. Looking forward to contributing on something else.